superset/views.py

```diff
@@ -1301,9 +1301,14 @@ def update_explore(self, datasource_type, datasource_id):
                 datasource_id=datasource_id,
                 args=form_data)
         except Exception as e:
-            flash('{}'.format(e), "alert")
-            return redirect(error_redirect)
-        return viz_obj.get_json()
+            logging.exception(e)
+            return json_error_response('{}'.format(e))
+        try:
+            payload = viz_obj.get_json()
+        except Exception as e:
+            logging.exception(e)
+            return json_error_response(utils.error_msg_from_exception(e))
+        return payload
```

Comment:
payload may be out of scope here, maybe the try block should `return viz_obj.get_json()`